core/src/main/scala/kafka/server/KafkaServer.scala

```diff
@@ -218,19 +218,17 @@ class KafkaServer(val config: KafkaConfig, time: Time = Time.SYSTEM, threadNameP
         this.logIdent = logContext.logPrefix
 
         /* create and configure metrics */
-        val brokerMetricsReporters = new DynamicMetricsReporters(brokerId, config.dynamicConfig)
         val reporters = new util.ArrayList[MetricsReporter]
-        reporters.addAll(brokerMetricsReporters.currentMetricsReporters.asJava)
         reporters.add(new JmxReporter(jmxPrefix))
         val metricConfig = KafkaServer.metricConfig(config)
         metrics = new Metrics(metricConfig, reporters, time, true)
-        brokerMetricsReporters.initMetrics(metrics)
+        new DynamicMetricsReporters(brokerId, config.dynamicConfig, metrics)
```

Comment:
This looks a little weird because the dynamic config is updated inside the constructor. However, it seems like we could just construct `DynamicMetricsReporters` inside `DynamicBrokerConfig.addReconfigurables` since it already has a reference to the brokerId and we can get `Metrics` from the `KafkaServer` instance.